certbot-nginx/certbot_nginx/tls_sni_01.py

```diff
@@ -63,7 +66,17 @@ def perform(self):
             if vhost.addrs:
                 addresses.append(list(vhost.addrs))
             else:
-                addresses.append([obj.Addr.fromstring(default_addr)])
+                if ipv6info[0]:
```

Comment:
Actually, what happens if we listen on both IPv6 and IPv4 even if IPv6 isn't enabled in Nginx or for the site? Does it crash? Or does it still work? Why only listen on IPv6 if the config is already doing so, for the default?

Reply:
Nginx will not start if listen [::] is defined and it's built without IPv6 support. However we're only listening IPv6 if the originating server block had IPv6 configuration (so we would have caught the problem already at the configtest). We need to add it here for port 443 too to be able to answer the TLS-SNI challenge.